### PR TITLE
[Fox] Restore Adobe Pass auth

### PR DESCRIPTION
**Pull request by gkoelln**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Restoring Adobe Pass auth, as well as adding additional metadata extraction information.

### CODE REVIEW

**Comment by raleeper** [comment]
This code worked for me. @dstftw I will forward a provider account to your email.

File: youtube_dl/extractor/fox.py [hunk 1]
```diff
@@ -68,21 +80,14 @@ def _real_extract(self, url):
         timestamp = unified_timestamp(video.get('datePublished'))
         age_limit = parse_age_limit(video.get('contentRating'))
 
-        data = try_get(
-            video, lambda x: x['trackingData']['properties'], dict) or {}
-
         creator = data.get('brand') or data.get('network') or video.get('network')
 
-        series = video.get('seriesName') or data.get(
+        series = video.get('seriesName') or video.get('series') or data.get(
```
Comment (dstftw) [inline youtube_dl/extractor/fox.py]: No.

File: youtube_dl/extractor/fox.py [hunk 2]
```diff
@@ -68,21 +80,14 @@ def _real_extract(self, url):
...
             'seriesName') or data.get('show')
-        season_number = int_or_none(video.get('seasonNumber'))
+        season_number = int_or_none(video.get('seasonNumber') or video.get('season_number'))
```
Comment (dstftw) [inline youtube_dl/extractor/fox.py]: No such field.

File: youtube_dl/extractor/fox.py [hunk 3]
```diff
@@ -68,21 +80,14 @@ def _real_extract(self, url):
...
         episode = video.get('name')
-        episode_number = int_or_none(video.get('episodeNumber'))
-        release_year = int_or_none(video.get('releaseYear'))
-
-        if data.get('authRequired'):
-            # TODO: AP
-            pass
+        episode_number = int_or_none(video.get('episodeNumber') or video.get('episode_number'))
```
Comment (dstftw) [inline youtube_dl/extractor/fox.py]: No such field.

File: youtube_dl/extractor/fox.py [hunk 4]
```diff
@@ -68,21 +80,14 @@ def _real_extract(self, url):
...
+        release_year = int_or_none(video.get('releaseYear') or video.get('release_year'))
```
Comment (dstftw) [inline youtube_dl/extractor/fox.py]: No such field.

**Comment by cyperous** [comment]
Is this going to be fixed anytime soon, I currently can't download anything from fox that requires a login.

**Comment by gkoelln** [comment]
@dstftw 

I don't know what you're seeing, but here's what I'm seeing for the fields in question (asterisks added to fields in question):

```
PS C:\WINDOWS\system32> youtube-dl https://www.fox.com/watch/0721b828c4948036e53e65649c55ee41/#playlist=59cb1d95eef1f900
1e097c65 -i --print-json -s -v
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['https://www.fox.com/watch/0721b828c4948036e53e65649c55ee41/#playlist=59cb1d95eef1f9001e097c
65', '-i', '--print-json', '-s', '-v']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2017.10.01
[debug] Python version 3.4.4 - Windows-10-10.0.15063
[debug] exe versions: ffmpeg 3.3.3, ffprobe 3.3.3, rtmpdump 2.3
[debug] Proxy map: {}
[debug] Default format spec: bestvideo+bestaudio/best
{
	"manifest_url": "https://content-ause1.uplynk.com/preplay2/6356c433ab9a4a66a896c7ff9ed8c996/dabd083f6dd4c987c956759536480bd0/PRSWtDr5mP3FTLKA1KCoP3czRXJv5OX7l4M11WfHyBI.m3u8?pbs=115deb56302b4222bbe51a29654ca58d",
	"requested_subtitles": null,
	"**release_year**": null,
	"format": "hls-5191 - 1920x1080",
	"playlist_index": null,
	"id": "0721b828c4948036e53e65649c55ee41",
	"http_headers": {
		"Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
		"Cookie": "gt_cks_0=\"1\\054b85b8b7cb0ddcec895958c4487ea56e516616501\\054eNplj08LgkAQxb+KLHiLbWd1/9hNIig81aVDhZiuJmXCahq09dlbUDCI4Q3D8Ob9mBdK6/paqgYtnAPCqlNaNW2r0gu+q9YQM27iIu6oKeLmoXOly+yzz6Id34ZhmPab5zqJrJWYv3uwNbciBpgnGZHgCUPQzEE47ytdjZC40+aIgBEeCGotGIBl6gyEcUxtF9IXrlhRAM6FCOzo0uURDcwp6JdGAsb9gXayuFvStFWd2S8nipTs/QUh30/L\"",
		"Accept-Language": "en-us,en;q=0.5",
		"Accept-Charset": "ISO-8859-1,utf-8;q=0.7,*;q=0.7",
		"Accept-Encoding": "gzip, deflate",
		"User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:10.0) Gecko/20150101 Firefox/47.0 (Chrome)"
	},
	"fulltitle": "The First Six Minutes",
	"duration": 350,
	"webpage_url": "https://www.fox.com/watch/0721b828c4948036e53e65649c55ee41/#playlist=59cb1d95eef1f9001e097c65",
	"**episode_number**": null,
	"_filename": "The First Six Minutes-0721b828c4948036e53e65649c55ee41.mp4",
	"description": "Watch the first six minutesof THE GIFTED, episode one.",
	"acodec": "mp4a.40.5",
	"url": "https://content-ause1.uplynk.com/6356c433ab9a4a66a896c7ff9ed8c996/k.m3u8?pbs=115deb56302b4222bbe51a29654ca58d",
	"height": 1080,
	"formats": [<Formats omitted for brevity>],
	"display_id": "0721b828c4948036e53e65649c55ee41",
	"upload_date": "20170927",
	"timestamp": 1506552900,
	"tbr": 5191.156,
	"title": "The First Six Minutes",
	"**season_number**": null,
	"ext": "mp4",
	"extractor_key": "FOX",
	"preference": null,
	"creator": "FOX",
	"width": 1920,
	"format_id": "hls-5191",
	"**series**": "The Gifted",
	"episode": "The First Six Minutes",
	"playlist": null,
	"vcodec": "avc1.640028",
	"age_limit": null,
	"webpage_url_basename": "0721b828c4948036e53e65649c55ee41",
	"protocol": "m3u8_native",
	"fps": 30,
	"extractor": "FOX"
}
```

**Comment by dstftw** [comment]
So what? Your "fallbacks" are not reached. Remove first value and check what you're extracting.

**Pull request closed without merging**